### A) F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

##### Minitab File : Cutlets.mtw


 #### Ho = There is no significance difference between diameter of the Culets
 #### H1 = There is significance difference between diameter of the Culets

In [1]:
import pandas as pd
import scipy
import numpy as np
from scipy import stats

In [2]:
UnitA = pd.Series([6.809,6.4376,6.9157,7.3012,7.4488,7.3871,6.8755,7.0621,6.684,6.8236,7.393,7.5169,6.9246,6.9256,6.5797,6.8394,6.597,7.2705,7.2828,7.3495,6.9438,7.156,6.5341,7.2854,6.9952,6.8568,7.2163,6.6801,6.9431,7.0852,6.7794,7.2783,7.1561,7.3943,6.9405])

In [3]:
UnitB = pd.Series([6.7703,7.5093,6.73,6.7878,7.1522,6.811,7.2212,6.6606,7.2402,7.0503,6.881,7.4059,6.7652,6.038,7.1581,7.024,6.6672,7.4314,7.307,6.7478,6.8889,7.422,6.5217,7.1688,6.7594,6.9399,7.0133,6.9182,6.3346,7.5459,7.0992,7.118,6.6965,6.578,7.3875])

In [4]:
stats.ttest_ind(UnitA,UnitB)

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [5]:
'''Since p value is greater than 5% significance hence we shall fail to reject Null Hypothesis , i.e There is no significance difference between diameter of the Culets'''

'Since p value is greater than 5% significance hence we shall fail to reject Null Hypothesis , i.e There is no significance difference between diameter of the Culets'

### B) A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 
 
    Minitab File: LabTAT.mtw


In [6]:
H0 =There is no significance difference between TAT of the laboratories
H1 =There is a significance difference between TAT of the laboratories

SyntaxError: invalid syntax (Temp/ipykernel_2856/466409188.py, line 1)

In [7]:
df=pd.read_csv("C:\\Users\\DELL\\Desktop\\assignments\\LabTAT.csv")

In [8]:
df

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [9]:
stats.f_oneway(df.iloc[:,0],df.iloc[:,1],df.iloc[:,2])

F_onewayResult(statistic=81.055779144798, pvalue=9.489467524356218e-30)

since the p-value here is greater than 0.05, we shall fail to reject Null Hypothesis i.e There is no significance difference between TAT of the laboratories

### C) Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.
1. Check p-value
2.If p-Value < alpha, we reject Null Hypothesis



### H0 = The male-female buyer rations are similar across regions
### H1 = The male-female buyer rations are not similar across regions

In [10]:
chisq_prob=pd.DataFrame([[50,142,131,70],[435,1523,1356,750]],
                        index=["Males","Females"],
                        columns=["East","West","North","South"])
chisq_prob

East  West  North  South
Males      50   142    131     70
Females   435  1523   1356    750

In [11]:
stats.chi2_contingency(chisq_prob)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [12]:
# to get the p-value
stats.chi2_contingency(chisq_prob)[1]

0.6603094907091882

Since p-value is greater than 0.05, we shall fail to reject Null Hypothesis, i.e The male-female buyer rations are similar across regions

### D) TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Minitab File: CustomerOrderForm.mtw


### H0 = The defective % does not varies by centre
### H1 = The defective % does varies by centr

In [13]:
df=pd.read_csv("C:\\Users\\DELL\\Desktop\\assignments\\Costomer+OrderForm.csv")

In [14]:
df

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [15]:
stats.chi2_contingency([df['Phillippines'].value_counts(), df['Indonesia'].value_counts(), df['Malta'].value_counts(), df['India'].value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

since p value which is 0.277 is greater than 0.05 hence we fail to reject Null Hypothesis ,i.e The defective % does not varies by centre